File: Exam2019s/Exam/PublicApi.v1/StudentHomeWorkDTO.cs
using System;

namespace PublicApi.v1
{
    public class StudentHomeWorkDTO
    {
        public Guid SubjectId { get; set; } = default!;
        public Guid HomeWorkId { get; set; } = default!;

        public bool IsAccepted { get; set; }
        public bool IsChecked { get; set; }

        public DateTime? Deadline { get; set; }
        public int Grade { get; set; }
    }

    public class StudentHomeWorkDetailsDTO
    {
        public string Title { get; set; } = default!;
        public string? Description { get; set; }
        public DateTime? Deadline { get; set; }

        public string SubjectTitle { get; set; } = default!;
        public string SubjectCode { get; set; } = default!;

        public Guid SubjectId { get; set; } = default!;
        public Guid HomeWorkId { get; set; } = default!;

        public bool IsAccepted { get; set; }
        public bool IsChecked { get; set; }

        public string? StudentAnswer { get; set; }
        public DateTime? AnswerDateTime { get; set; }

        public int Grade { get; set; }
    }
}

File: Exam2019s/exam-aurelia/src/types/StudentHomeworks/StudentHomeworkDTO.ts
export interface StudentHomeworkDTO {
    id: string;
    title: string;
    description: string;
    deadline: Date;
    subjectTitle: string;
    subjectCode: string;
    subjectId: string;
    homeWorkId: string;
    isAccepted: boolean;
    isChecked: boolean;
    studentAnswer: string;
    answerDateTime: Date;
    grade: number;
}

export interface StudentHomeworkPostDTO {
    studentAnswer?: string;
    studentSubjectId: string;
    homeWorkId: string;
}

export interface StudentHomeworkPutDTO {
    id: string;
    studentAnswer?: string;
}

export interface StudentHomeworkTeacherSubmitDTO {
    id: string;
    isAccepted: boolean;
    isChecked: boolean;
    grade: number;
}
